### PR TITLE
feat: store deployment key for deployed forms

### LINKED ISSUE #19709: Store reference to deployment for deployed resources
### Description
To support the new [`deployment` binding type](https://github.com/camunda/camunda/issues/19384), Zeebe needs to be able to tell which resources (processes, decisions, forms) were deployed together in the same deployment. Therefore, we should store a reference to the deployment for each deployed resource.

### Acceptance Criteria
- [x] When a deployment request is processed and the contained resources are transformed, the `deploymentKey` is stored as a new property in:
  - [x] `ProcessRecord` (for all processes in a BPMN resource)
  - [x] `DecisionRecord` (for all decisions in a DMN resource)
    - _Note_: It doesn't seem to be necessary to store the deployment key in a `DecisionRequirementsRecord` as well, because only the decision ID is relevant when `deployment` binding is used in a business rule task
  - [x] `FormRecord` (for form resources)
- [x] When the records are persisted in the state, the `deploymentKey` is stored as a new property in:
  - [x] `PersistedProcess`
  - [x] `PersistedDecision`
  - [x] `PersistedForm`
- [x] New column families are added to store:
  - [x] Processes by `processId` and `deploymentKey`
  - [x] Decisions by `decisionId` and `deploymentKey`
  - [x] Forms by `formId` and `deploymentKey`
  - _Note_: To avoid taking up additional storage space, the column families could just store a foreign key (e.g. `processDefinitionKey`) instead of the whole object.
- [x] New methods are added to the state classes to look up the objects from the new column families
(these will be used in #19710)
  - The methods should use a cache to reduce latency.
- [x] The new column families are only filled for new deployments to ensure existing events are replayed deterministically. This should be done by introduction new record and event applier versions.

### Dependencies
- required by #19710

### PR DESCRIPTION
**Pull request by jfriedenstab**
## Description
Store the deployment key as a new property for deployed forms and add the possibility to find a form by form id and deployment key. This is a prerequisite to be able to properly resolve linked forms when the new `deployment` binding type is used (see #19710).

## Related issues
Closes #19709 (processes were already handled with #20498; decisions with #20628)

### CODE REVIEW

File: zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java [hunk 1]
```diff
@@ -171,7 +171,8 @@ public enum ZbColumnFamilies implements EnumValue {
   USER_TASK_STATES(80),
   COMPENSATION_SUBSCRIPTION(81),
 
-  PROCESS_DEFINITION_KEY_BY_PROCESS_ID_AND_DEPLOYMENT_KEY(82);
+  PROCESS_DEFINITION_KEY_BY_PROCESS_ID_AND_DEPLOYMENT_KEY(82),
+  FORM_KEY_BY_FORM_ID_AND_DEPLOYMENT_KEY(84);
```
Comment (jfriedenstab) [inline zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java:175]: 83 was added with #20628.

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/FormResourceTransformer.java [hunk 2]
```diff
@@ -131,14 +130,16 @@ private void appendMetadataToFormRecord(
       final FormMetadataRecord formRecord,
       final String formId,
       final DeploymentResource resource,
-      final String tenantId) {
+      final DeploymentRecord deployment) {
```
Comment (korthout) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/FormResourceTransformer.java:133]: 👍 I like it

**Review by korthout — APPROVED** [review]
Thanks @jfriedenstab 💯 

LGTM 👍

**Pull request merged**